pgspecial/dbcommands.py

```diff
@@ -1310,3 +1311,40 @@ def sql_name_pattern(pattern):
         schema = '^(' + schema + ')$'
 
     return schema, relname
+
+
+@special_command('\\sf', '\\sf[+] FUNCNAME', 'Show a function\'s definition.', hidden=True)
+def show_function_definition(cur, pattern, verbose):
+    if '(' in pattern:
+        sql = cur.mogrify("SELECT %s::pg_catalog.regprocedure::pg_catalog.oid", [pattern])
+    else:
+        sql = cur.mogrify("SELECT %s::pg_catalog.regproc::pg_catalog.oid", [pattern])
+    log.debug(sql)
+    try:
+        cur.execute(sql)
+    except psycopg2.ProgrammingError as e:
```

Comment:
Why do we handle this specific `cur.execute()` error here instead of letting it bubble up to pgcli? 

Looks like everywhere else we just run `cur.execute()` without a try/except around it. 

The reason I ask is that I'm not sure I'd like to add `psycopg2` as a dependency to this library without an absolute necessity.